### PR TITLE
Pratik

### PR DESCRIPTION
**Pull request by pratiksatham**
Please Review My Code

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+from flask import Flask, jsonify, request
+import mysql.connector
+from flask_cors import CORS,cross_origin
+import bcrypt
+from flask_bcrypt import bcrypt
+from connection import *
+from queries import *
+from pmt import *
+import logging
+
```
Comment (Simran2711) [inline app.py:10]: please remove unnecessary spaces.

File: app.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+from flask_bcrypt import Bcrypt
+import bcrypt
+bcrypt = Bcrypt(app)
+
+
+@app.route('/login', methods=['POST'])
+def pm_login(): 
+    return pm_loginn()
+
+
```
Comment (Simran2711) [inline app.py:34]: Here also remove spaces.

File: connection.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+from flask import jsonify
+from flask import Flask, jsonify, request
+import mysql.connector
+from urllib.parse import quote_plus as urlquote
+
+
```
Comment (Simran2711) [inline connection.py:6]: Please add proper headings.

File: pmt.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+    except KeyError:
+        # Handle missing key error
+        return jsonify({'error': 'Missing key in the request'}), 400
+
+    except Exception as e:
+        # Handle other exceptions
+        return jsonify({'error': str(e)}), 500
+
+
+
```
Comment (Simran2711) [inline pmt.py:154]: Give headings of Task Module

File: queries.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
+from connection import *
+# MySQL configuration
+
+
+mydb=connect_db()
+cursor=mydb.cursor()
+
+
+
```
Comment (Simran2711) [inline queries.py:9]: Remove the add User Function

File: pmt.py [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+logging.basicConfig(level=logging.DEBUG)
+
+file = open("myfile.txt","w")
+
+app = Flask(__name__)
+cors = CORS(app)
+
+
+
+##############################################################################################################
```
Comment (Simran2711) [inline pmt.py:21]: Remove this heading of Login Details and Add the Description Instead

File: pmt.py [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+        return jsonify({"error": "Missing key in request data: " + str(e)}), 400
+
+    except Exception as e:
+        # Handle any other unexpected exceptions
+        logging.error("An error occurred: " + str(e))
+        return jsonify({"error": "An error occurred: " + str(e)}), 500
+    
+
+
+############################ DELETE TASK #################################
```
Comment (Simran2711) [inline pmt.py:218]: Remove the headings

File: pmt.py [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+        logging.error("Missing key in request data: " + str(e))
+        return jsonify({"error": "Missing key in request data: " + str(e)}), 400
+
+    except Exception as e:
+        # Handle any other unexpected exceptions
+        logging.error("An error occurred: " + str(e))
+        return jsonify({"error": "An error occurred: " + str(e)}), 500
+    
+
+
```
Comment (Simran2711) [inline pmt.py:239]: Remove the Unwanted Spaces.

**Review by Simran2711 — CHANGES_REQUESTED** [review]
Please do the requested changes

File: app.py (hunk 1 above)
Comment (pratiksatham) [reply app.py]: Done

Comment (pratiksatham) [reply app.py]: Ok i did this

File: app.py (hunk 2 above)
Comment (pratiksatham) [reply app.py:31]: ok i did this

File: connection.py (hunk 3 above)
Comment (pratiksatham) [reply connection.py:6]: i fixed this please check this again

File: queries.py (hunk 5 above)
Comment (pratiksatham) [reply queries.py:9]: i fixed this, please review again

File: app.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+logging.basicConfig(level=logging.DEBUG)
+
+file = open("myfile.txt","w")
+
+app = Flask(__name__)
+cors = CORS(app)
+
+
```
Comment (Srishtirupa3) [inline app.py:17]: Please Remove this space

File: pmt.py [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+
+##############################################################################################################
+                                        # login
+###############################################################################################################
+
+
+def pm_loginn():
+    try:
```
Comment (Srishtirupa3) [inline pmt.py:27]: Please add doc string

File: app.py (hunk 9 above)
Comment (pratiksatham) [reply app.py:17]: i have done this

**Review by Chandrashekhar-Bhavsar — APPROVED** [review]

**Review by Simran2711 — APPROVED** [review]

**Review by Chandrashekhar-Bhavsar — CHANGES_REQUESTED** [review]
Add Unit test file

File: pmt.py [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+
+    except KeyError:
+        # Handle missing key error
+        return jsonify({'error': 'Missing key in the request'}), 400
+
+    except Exception as e:
+        # Handle other exceptions
+        return jsonify({'error': str(e)}), 500
+
+
```
[the remark this answers is not in the document]
Comment (pratiksatham) [reply pmt.py:153]: added

File: pmt.py (hunk 7 above)
Comment (pratiksatham) [reply pmt.py:218]: removed

File: pmt.py (hunk 8 above)
Comment (pratiksatham) [reply pmt.py:239]: removed

**Review by pratiksatham — COMMENTED** [review]
i have made all the changes please check it out

File: queries.py [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+        cursor = mydb.cursor()
+        query = "INSERT INTO project_details (Project_Name, Project_Description, Planned_SD, Planned_ED,Actual_SD, Actual_ED, Planned_Hours, Actual_Hours, Status, Project_Lead, Client_Name, Risk, Mitigation) VALUES ( %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)"
+        values = (project_name, project_description, planned_sd, planned_ed, actual_sd, actual_ed,
+                  planned_hours, actual_hours, status, project_lead, client_name, risk, mitigation)
+        cursor.execute(query, values)
+        mydb.commit()
+
+        return jsonify({"message": "Project created successfully"}), 200
+
+
```
Comment (Chandrashekhar-Bhavsar) [inline queries.py:35]: Uniform spacing

**Review by Chandrashekhar-Bhavsar — APPROVED** [review]
Unit test file added good to go

**Review by Chandrashekhar-Bhavsar — APPROVED** [review]
Unit test file added good to go

**Review by Chandrashekhar-Bhavsar — APPROVED** [review]
Unit test file added good to go

**Review by Srishtirupa3 — APPROVED** [review]

**Pull request closed without merging**